x-pack/plugins/apm/public/components/app/Settings/anomaly_detection/update_jobs_callout.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiButton, EuiCallOut } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { callApmApi } from '../../../../services/rest/createCallApmApi';
+
+export function UpdateJobsCallout({
+  onUpdateComplete,
+}: {
+  onUpdateComplete: () => void;
+}) {
+  const [loading, setLoading] = useState(false);
+  const { core } = useApmPluginContext();
+
+  function upgradeJobs() {
+    setLoading(true);
+    callApmApi({
+      endpoint: 'POST /internal/apm/settings/anomaly-detection/update_to_v3',
+      signal: null,
+    })
+      .then(() => {
+        core.notifications.toasts.addSuccess({
+          title: i18n.translate(
+            'xpack.apm.jobsList.updateCompletedToastTitle',
+            {
+              defaultMessage: 'Anomaly detection jobs created!',
+            }
+          ),
+          text: i18n.translate('xpack.apm.jobsList.updateCompletedToastText', {
+            defaultMessage:
+              'Your new anomaly detection jobs have been created successfully. You will start to see anomaly detection results in the app within minutes. The old jobs have been closed but the results are still available within Machine Learning.',
+          }),
+        });
+        onUpdateComplete();
+      })
+      .catch(() => {
+        setLoading(false);
+      });
+  }
+
+  return (
+    <EuiCallOut
+      color="success"
+      title={i18n.translate(
+        'xpack.apm.settings.anomalyDetection.jobsList.updateAvailableTitle',
+        { defaultMessage: 'Updates available' }
+      )}
+      iconType="wrench"
+    >
+      <p>
+        {i18n.translate(
+          'xpack.apm.settings.anomalyDetection.jobsList.updateAvailableDescription',
+          {
+            defaultMessage:
+              'We have updated the anomaly detection jobs that provide insights into degraded performance and added detectors for throughput and failed transaction rate. If you choose to upgrade, we will create the new jobs and close the existing legacy jobs. The data shown in the APM app will automatically switch to the new.',
```

Comment:
It's not super clear why the user should upgrade, or what the consequences of not updating.

Perhaps we should mention transaction metrics directly:

> We have detected that you are using transaction metrics, which is not supported by your current ML jobs. Please upgrade to ensure your ML jobs work correctly. If you choose to upgrade, we will create the new jobs and close the existing legacy jobs. Your data from the legacy jobs will still be available to view in the ML app. 

(wording can be improved)

Reply:
IMO, I think the upgrade message clearly describes why they should upgrade and what happens after. We intentionally made the message more about the benefits of upgrading rather than trying to explain the technical aspects. I think your proposal will only make it more confusing to understand the why.

Reply:
I agree that it's better to keep the implementation details away from the end user. Still, when reading the current message it feels like an optional upgrade, and not something we strongly urge users to do. ML team said that even if  all transactions are sampled, they still wouldn't recommend showing transaction-based ML results on top of metric based data (which is what will happen until people upgrade afaik). 

Either way, this will be easy to change down the road so we can release it as-is and adapt if we get feedback.

Reply:
> ML team said that even if all transactions are sampled, they still wouldn't recommend showing transaction-based ML results on top of metric based data (which is what will happen until people upgrade afaik).

That information wasn't available at the time when we made the choice to design the migration this way. If that's the case, I'd recommend us looking into a more strict migration in the next iteration. That could still just involve us changing the messaging and displays to say that we're not going to show the legacy job results in the app but only rely on the new jobs results. Do you think it's worth opening a new issue to track this?

Reply:
I went back to see where ML said that we shouldn't mix metric events and transaction anomalies and found this from @sophiec20 :

> think we would be safer with transaction charts showing transaction anomalies, and metric powered charts showing metric anomalies. The anomalies would probably be OK, but the expected bounds might be very different.

It's then followed up with an edit (that I had missed) saying:

> expected bounds are for mean transaction duration, therefore it should be ok to plot providing sampling is representative

The way I read this we are okay to mix as long as end users sampling is representative.

Reply:
@dgieselaar quick question - is the ML job version tracked in our current telemetry? Just wondering if we can employ telemetry to figure out if our migration rate is too slow and then try another strategy to get users to convert. I also think once we will display expected bounds for all metrics, it'll make more sense to get jobs upgraded.

Reply:
it's not, we would have to add it, which would require us to migrate to telemetry v2 first

Reply:
OK, so we should create a new issue to implement it even if we do it in another iteration. I think the telemetry would be nice to have 👍